pyomo/contrib/mindtpy/tests/eight_process_problem.py

```diff
@@ -0,0 +1,149 @@
+"""Re-implementation of eight-process problem.
+
+Re-implementation of Duran example 3 superstructure synthesis problem in Pyomo
+Author: Qi Chen <https://github.com/qtothec>.
+
+Ref:
+    SELECT OPTIMAL PROCESS FROM WITHIN GIVEN SUPERSTRUCTURE.
+    MARCO DURAN , PH.D. THESIS (EX3) , 1984.
+    CARNEGIE-MELLON UNIVERSITY , PITTSBURGH , PA.
+
+The expected optimal solution value is 68.0.
+
+(original problem, my implementation may vary)
+    Problem type:    convex MINLP
+            size:    8  binary variables
+                    26  continuous variables
+                    32  constraints
+
+Pictoral representation can be found on
+page 969 of Turkay & Grossmann, 1996.
+http://dx.doi.org/10.1016/0098-1354(95)00219-7
+
+"""
+from __future__ import division
+
+from six import iteritems
+
+from pyomo.environ import (Binary, ConcreteModel, Constraint, NonNegativeReals,
+                           Objective, Param, RangeSet, Var, exp, minimize)
+
+
+class EightProcessFlowsheet(ConcreteModel):
```

Comment:
Should this be replaced by just importing the GDP model and calling `bigM` on it, or is it worth keeping around the MINLP reformulation separately?

Reply:
I would go for the independent files for the moment, but I'm open to just use the Big-M transformation of the GDP.